Assets/Scripts/Models/RoomManager.cs

```diff
@@ -0,0 +1,354 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+public class RoomManager: IEnumerable<Room>
+{
+    public event Action<Room> Adding;
+
+    public event Action<Room> Added;
+
+    public event Action<Room, Room> Split;
+
+    public event Action<Room, Room> Joined;
+
+    public event Action<Room> Removing;
+
+    public event Action<Room> Removed;
+
+    private List<Room> rooms;
+
+    public RoomManager()
+    {
+        rooms = new List<Room>();
+    }
+
+    public int Rooms {
+        get { 
+            return rooms.Count;
+        }
+    }
+
+    public Room GetOutsideRoom()
+    {
+        // TODO maybe find some other way of knowing what the outside is?
+        // or perhaps have it not just be a 0 here, maybe a const?
+        return rooms[0];
```

Comment:
Maybe the RoomManager should create the outside room in the constructor instead of assuming that it always have a Room at index 0 and that that one is the outside one.